drivers/sx1276/sx1276.c

```diff
@@ -0,0 +1,1303 @@
+                        break;
+                }
+            }
+            else {
+                sx1276_reg_write(dev, SX1276_REG_LR_DETECTOPTIMIZE,
+                                 sx1276_reg_read(dev, SX1276_REG_LR_DETECTOPTIMIZE) | 0x80);
+            }
+
+            rx_continuous = dev->settings.lora.rx_continuous;
+
+            /* Setup interrupts */
+            if (dev->settings.lora.freq_hop_on) {
+                sx1276_reg_write(dev, SX1276_REG_LR_IRQFLAGSMASK,  
+									/* SX1276_RF_LORA_IRQFLAGS_RXTIMEOUT |
+		                            SX1276_RF_LORA_IRQFLAGS_RXDONE |
+		                            SX1276_RF_LORA_IRQFLAGS_PAYLOADCRCERROR | */
+                                 SX1276_RF_LORA_IRQFLAGS_VALIDHEADER |
+                                 SX1276_RF_LORA_IRQFLAGS_TXDONE |
+                                 SX1276_RF_LORA_IRQFLAGS_CADDONE |
+                                 	/* SX1276_RF_LORA_IRQFLAGS_FHSSCHANGEDCHANNEL | */
+                                 SX1276_RF_LORA_IRQFLAGS_CADDETECTED);
+
+                /* DIO0=RxDone, DIO2=FhssChangeChannel */
+                sx1276_reg_write(dev,
+                                 SX1276_REG_DIOMAPPING1,
+                                 (sx1276_reg_read(dev, SX1276_REG_DIOMAPPING1)
+                                  & SX1276_RF_LORA_DIOMAPPING1_DIO0_MASK
+                                  & SX1276_RF_LORA_DIOMAPPING1_DIO2_MASK)
+                                 | SX1276_RF_LORA_DIOMAPPING1_DIO0_00
+                                 | SX1276_RF_LORA_DIOMAPPING1_DIO2_00);
+            }
+            else {
+                sx1276_reg_write(dev, SX1276_REG_LR_IRQFLAGSMASK,  
+										/* SX1276_RF_LORA_IRQFLAGS_RXTIMEOUT |
+											SX1276_RF_LORA_IRQFLAGS_RXDONE |
+											SX1276_RF_LORA_IRQFLAGS_PAYLOADCRCERROR | */
+                                 SX1276_RF_LORA_IRQFLAGS_VALIDHEADER |
+                                 SX1276_RF_LORA_IRQFLAGS_TXDONE |
+                                 SX1276_RF_LORA_IRQFLAGS_CADDONE |
+                                 SX1276_RF_LORA_IRQFLAGS_FHSSCHANGEDCHANNEL |
+                                 SX1276_RF_LORA_IRQFLAGS_CADDETECTED);
+
+                /* DIO0=RxDone */
+                sx1276_reg_write(dev,
+                                 SX1276_REG_DIOMAPPING1,
+                                 (sx1276_reg_read(dev, SX1276_REG_DIOMAPPING1)
+                                  & SX1276_RF_LORA_DIOMAPPING1_DIO0_MASK)
+                                 | SX1276_RF_LORA_DIOMAPPING1_DIO0_00);
+            }
+
+            sx1276_reg_write(dev, SX1276_REG_LR_FIFORXBASEADDR, 0);
+            sx1276_reg_write(dev, SX1276_REG_LR_FIFOADDRPTR, 0);
+        }
+        break;
+    }
+
+    sx1276_set_status(dev, SX1276_RF_RX_RUNNING);
+
+    if (rx_continuous) {
+        sx1276_set_op_mode(dev, SX1276_RF_LORA_OPMODE_RECEIVER);
+    }
+    else {
+        if (timeout != 0) {
+            xtimer_set(&(dev->_internal.rx_timeout_timer), timeout);
```

Comment:
BTW the [netdev2 adoption] is ready. I'm currently adapting the options to NETOPT format.

[netdev2 adoption]:https://github.com/Inria-Chile/RIOT/tree/sx1276-netdev2/drivers/sx1276